### DISCUSSION TITLE
Add support for distance per stroke

### DISCUSSION THREAD

**Discussion by raju-bitter** [time4tea/gopro-dashboard-overlay#209] [Ideas]
First, let me thank you for creating this software, it's fantastic!
I volunteer to add support for a "distance per stroke" metric. Distance per stroke is used in sports like swimming, standup paddling, rowing, and kayaking. I am a kayaker / surfski paddler, and use the Vaaka cadence sensor to record my cadence and distance per stroke.
https://www.vaakacadence.com/

Vaaka has a Garmin IQ Connect app for recording the cadence and distance per stroke metric when paddling: https://apps.garmin.com/en-US/apps/b30d0330-ad62-4a32-aefb-93d5bb071c03

In my Garmin Connect web UI I can see the distance per stroke chart, which is calculated based on the GPS data.
![connect garmin com_modern_activity_16964358166 (1)](https://github.com/user-attachments/assets/2738b42d-b9ca-44ab-8b97-dd3b69e36bec)

Distance per stroke is a useful metric when tracking training improvement for paddlers, swimmers, or rowers.
Garmin Fenix 7 docs for the stroke field: https://www8.garmin.com/manuals-apac/webhelp/fenix7series/EN-SG/GUID-1FB601E1-98B8-4D83-A19D-F2D1AD804339-7981.html

**Comment by time4tea** [comment]
Thank you for your kind words!

If the DPS metric is included in the .fit file then it is probably very easy to include.

Do you have a sample that you can share? If so, please send it to gopro-overlay@time4tea.net 

All submitted files are stored on encrypted systems.

If it is calculated, then added to the GPS file, then it could also be easy to add, and it would be super useful if you had a sample of that file.

**Reply by raju-bitter** [reply]
I sent you an example of a FIT file generated using the Vaaka cadence sensor. I will create another file using the standup paddle board profile, supported by the Garmin Fenix without additional sensors. For swimming I'm not sure if the same field is used for SUP paddling. I'll try to generate a file for swimming as well.

**Vaaka Sensor**
The data is in the FIT file, in the "records" section. The field is named "Sdps", and when using the Vaaka sensor the "Sport" entry uses the "sub sport" value "Vaaka paddling".

![image](https://github.com/user-attachments/assets/283d5b54-7402-4ec0-8667-3b11e4ed972a)

**Reply by time4tea** [reply]
Got it. Thank you!

**Reply by raju-bitter** [reply]
It is only a two line change to extract the information from the FIT file.
https://github.com/time4tea/gopro-dashboard-overlay/compare/main...raju-bitter:gopro-dashboard-overlay-dps:vaaka-distance-per-stroke

The layout XML snippet used for:
```
    <composite x="1900" y="820" name="distance_per_stroke">
        <component type="text" x="-70" y="0" size="16" align="right">DPS</component>
        <component type="metric" x="-70" y="18" metric="sdps" dp="2" size="32" align="right"/>
    </composite>
```
And the result:
![image](https://github.com/user-attachments/assets/7b22c0b9-118d-4484-9a8c-4ae9e1defe8d)

What kind of tests do you normally run for new value fields?

Here are my planned next steps:

1.  Add an automated test.
2. Check what the field name is for swimming and standup paddling, using the native Garmin implementation on the smartwatch.
3. Add an example and documentation.
